### ISSUE TITLE
[api-extractor] Update Typescript dependency?

### ISSUE THREAD

**Issue by ethanresnick** [Microsoft/web-build-tools#689]
I'm trying to use API Extractor, but getting an error because my `tsconfig.json` uses a compiler option not introduced until TS 2.7, whereas api-extractor currently depends on TS `~2.4.1`. Is there a reason that TS is fixed to this version?  If not, I'd be happy to open a PR updating to TS `2.9.x`. (Note: if I manually update package.json to depend on a newer version of Typescript, that doesn't seem to cause any problems on my code, at least.)

**Comment by pgonzal** [comment]
Yes, in the past API Extractor ran on the *.ts files instead of the *.d.ts files.  To make this efficient, we shared that same compiler/AST as the regular tool chain.  This coupled API Extractor to the compiler version that we were using for all our production projects, which is generally fairly outdated.  (Since there are hundreds of projects, migrating to a new compiler version is always an involved process that takes about a week of work, so we are lazy about upgrading.  Also, for platform libraries, using the latest compiler is undesirable because it blocks consumers from upgrading unless they upgrade their compiler.  To maximize compatibility, applications want the latest version anyone is using, whereas core libraries want the oldest version anyone is using.)

All this changed when we switched API Extractor to process the *.d.ts files.  It is now totally possible for API Extractor to be on the latest TypeScript compiler.  I'm planning to make that change as part of this month's sprint.

**Comment by ethanresnick** [comment]
@pgonzal Thanks for the thorough explanation! I'm looking forward to it.
